backend/app/controllers/api/v1/contacts_controller.rb

```diff
@@ -1,67 +1,60 @@
+
+    def index        
         contacts = Contact.all.order(created_at: :desc)
 
         render json: { contacts: contacts }, status: 200
     end
 
-    def show
-        return render json: { errors: '未ログイン' }, status: 401 unless api_v1_user_signed_in?
-
-        if params[:id].present?
-            begin
-                contact = Contact.find(params[:id].to_i)
-            rescue ActiveRecord::RecordNotFound
-                return render json: { errors: '対象のデータが見つかりません' }, status: 404
-            end
-        else
-            return render json: { errors: "対象の問い合わせデータがありません" }, status: 404
-        end
-
+    def show        
+        contact = Contact.find(params[:id].to_i)
         render json: { contact: contact }, status: 200
+    rescue ActiveRecord::RecordNotFound
+        return render json: { errors: '対象のデータが見つかりません' }, status: 404
+    rescue StandardError => e
+        render json: { errors: e.message }, status: 500
     end
 
-    def create
-        user = current_api_v1_user
-        
+    def create        
         contact = Contact.new(contact_register_params)
         if user.present?
-            contact.user_id = user.id
+            contact.user_id = current_api_v1_user.id
         end
 
-        if contact.save
-            ContactMailer.complete.deliver_later
-            render json: { contact: contact }, status: 200
-        else
-            render json: { errors: contact.errors.full_messages }, status: 422
-        end
+        render json: { errors: contact.errors.full_messages }, status: 422 and return if contact.invalid?
+
+        contact.save!
+
+        ContactMailer.complete.deliver_later
+
+        render json: { contact: contact }, status: 200
+    rescue StandardError => e
+        render json: { errors: e.message }, status: 500
     end
 
     def update
-        return render json: { errors: '未ログイン' }, status: 401 unless api_v1_user_signed_in?
-
-        if params[:id].present?
-            begin
-                contact = Contact.find(params[:id].to_i)
-            rescue ActiveRecord::RecordNotFound
-                return render json: { error: '対象のデータが見つかりません' }, status: 404
-            end
-        else
+        if params[:id].nil?
             return render json: { errors: "対象の問い合わせデータがありません" }, status: 404
         end
 
-        if contact.update(contact_register_params)
-            render json: { contact: contact }, status: 200
-        else
-            render json: { errors: contact.errors.full_messages}, status: 422
-        end
+        contact = Contact.find(params[:id].to_i)
+        contact.update(contact_register_params)
+
+        render json: { errors: contact.errors.full_messages }, status: 422 and return if contact.invalid?
+
+        contact.save!
+
+        render json: { contact: contact }, status: 200
+    rescue ActiveRecord::RecordNotFound
+        render json: { error: '対象のデータが見つかりません' }, status: 404
+    rescue StandardError => e
```

Comment:
contact.invalid?でバリデーションチェックをしてますが、
contact.update(contact_register_params)の時点で更新が走ってしまい、失敗した場合は例外でなくfalseを返すようになってます。
そのため以下のようにupdateをupdate!にして更新に失敗した場合に例外を発生させるようにしrescue ActiveRecord::RecordInvalid => eでバリデーションエラーをキャッチする形が自然かと思います！
```suggestion
        contact = Contact.find(params[:id].to_i)
        contact.update!(contact_register_params)

        render json: { contact: contact }, status: 200
    rescue ActiveRecord::RecordNotFound
        render json: { error: '対象のデータが見つかりません' }, status: 404
    rescue ActiveRecord::RecordInvalid => e
      render json: { errors: e.record.errors.full_messages }, status: 422
    rescue StandardError => e
```

Reply:
ActiveRecord::RecordInvalidでキャッチするように修正しました